### ISSUE TITLE
When oversubscribing on a system with multiple NIC, the PMIx call for locality string from local peers returns a null locality string. 

### ISSUE THREAD

**Issue by dancejic** [open-mpi/ompi#8236]
### What version of Open MPI are you using? (e.g., v3.0.5, v4.0.2, git branch name and hash, etc.)

commit 28779321a721b2202c9e6224070f729263266044 (HEAD -> master, origin/master, origin/HEAD)

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

git clone

### If you are building/installing from a git clone, please copy-n-paste the output from `git submodule status`.

 9da252a5ecb5fddb45f93a5169e6e4ef3b7280ec 3rd-party/openpmix (v1.1.3-2617-g9da252a5)
 ffdf6916eb600d0b5e37ef6747c5384944db8625 3rd-party/prrte (dev-30751-gffdf6916eb)

### Please describe the system on which you are running

* Operating system/version: 
```
NAME="Amazon Linux"
VERSION="2"
```
* Computer hardware: 
```
Architecture:        x86_64
Core(s) per socket:  24
Socket(s):           2
NIC: 4
```
* Network type: Single and multiple nodes

-----------------------------

## Details of the problem

When oversubscribing on a system with multiple NIC, the PMIx call for locality string from local peers fails. This causes the following output because the calculation of package rank in opal/mca/common/ofi/common_ofi.c is unable to proceed without the local peers locality string:
```
--------------------------------------------------------------------------
Open MPI's OFI driver detected multiple equidistant NICs from the current process,
but had insufficient information to ensure MPI processes fairly pick a NIC for use.
This may negatively impact performance. A more modern PMIx server is necessary to
resolve this issue.
--------------------------------------------------------------------------
```
this is the command I'm running:
```mpirun --map-by :OVERSUBSCRIBE --np 49 --mca btl ^openib ompi/examples/hello_c```

@rhc54 do you know why this might be happening? this happens here:
opal/mca/common/ofi/common_ofi.c:
```
// Get the LOCALITY_STRING for process[i]
OPAL_MODEX_RECV_VALUE_OPTIONAL(rc, PMIX_LOCALITY_STRING,
                               &pname, &locality_string, PMIX_STRING);
if (PMIX_SUCCESS != rc || NULL == locality_string) {
    // If we don't have information about locality, fall back to procid
    opal_show_help("help-common-ofi.txt", "package_rank failed", true);
    return (uint32_t)process_info->myprocid.rank;
}
```
It seems like the return status is success, but the locality_string ends up null:
```return: 0 locality: (null)```

**Comment by rhc54** [comment]
Hmmm...well, it shouldn't return "success" (I'll look and see why that happens), but the reason you aren't getting a locality is because you are oversubscribing the nodes. When you oversubscribe, we don't bind the procs - and so the procs don't have a locality on the node.

**Comment by rhc54** [comment]
FWIW: doesn't have anything to do with the number of NICs on the node.

**Comment by rajachan** [comment]
Yeah, based on what Ralph said, we that error message is incorrect when oversubscribing a node. As long as the fallback with procid works as expected, we should just not print that error in the oversubscription case (or more broadly, in cases where procs don't get bound to a core).

**Comment by dancejic** [comment]
Ok, I'll work on a fix to skip over the error message in case of oversubscribing.

**Comment by rhc54** [comment]
Just FYI: the MPI layer doesn't have a way of knowing if we are oversubscribed or not. I suspect you'll need to change that error message into a debug output (e.g., `opal_output_verbose`)

**Comment by dancejic** [comment]
Yeah I was realizing that as I was digging through trying to find a good way to determine that. I'll do that instead, thanks!

**Comment by dancejic** [comment]
I posted a PR to fix the error messages: #8244
